packages/admin-api/src/controllers/notification.controller.ts

```diff
@@ -0,0 +1,94 @@
+import express from "express"
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as notificationService from "../services/notification.service"
+import { getCatchments } from "../lib/catchment"
+
+export const getNotification = async (req: any, res: express.Response) => {
+    try {
+        const { catchmentNo, type } = req.query
+        const catchments = await getCatchments(req.kauth.grant.access_token)
+        if (catchments.length === 0 || !catchments.includes(Number(catchmentNo))) {
+            return res.status(403).send("Forbidden")
+        }
+        const { email } = req.kauth.grant.access_token.content
+        if (type !== "application" && type !== "claim") {
+            return res.status(400).send("Invalid type")
+        }
+        const username = `${req.kauth.grant.access_token.content.idp_username}@${req.kauth.grant.access_token.content.identity_provider}`
```

Comment:
Should add checks for the values extracted and used from the token, similar to the other controllers.